Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -208,22 +208,22 @@ void CookieStore::set(CookieInit&& options, Ref<DeferredPromise>&& promise)
         return;
     }
 
-    auto& url = document.url();
-    auto& cookieJar = page->cookieJar();
-    auto completionHandler = [promise = WTFMove(promise)] (bool setSuccessfully) {
-        if (!setSuccessfully)
-            promise->reject(TypeError);
-        else
-            promise->resolve();
-    };
-
     Cookie cookie;
     cookie.name = WTFMove(options.name);
     cookie.value = WTFMove(options.value);
     cookie.domain = options.domain.isNull() ? document.domain() : WTFMove(options.domain);
-    cookie.path = WTFMove(options.path);
     cookie.created = WallTime::now().secondsSinceEpoch().milliseconds();
 
+    cookie.path = WTFMove(options.path);
+    if (!cookie.path.isNull()) {
+        if (cookie.path.isEmpty() || cookie.path.characterAt(0) != '/') {
+            promise->reject(TypeError);
```

Comment:
Please add a description for the exception. How is the web dev supposed to know what there are doing wrong?